kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -86,13 +94,51 @@ private TableConfig(
      * @return the value of the table property
      */
     public T fromMetadata(Metadata metadata) {
-        T value = fromString.apply(metadata.getConfiguration().getOrDefault(key, defaultValue));
-        if (!validator.test(value)) {
+        String value = metadata.getConfiguration().getOrDefault(key, defaultValue);
+        validate(value);
+        return fromString.apply(value);
+    }
+
+    /**
+     * Returns the key of the table property.
+     *
+     * @return the key of the table property
+     */
+    public String getKey() {
+        return key;
+    }
+
+    /**
+     * Validates the given properties.
+     *
+     * @param configurations the properties to validate
+     *
+     * @throws IllegalArgumentException if any of the properties are invalid
```

Comment:
Looks like we are also throwing `UnknownConfigurationKeyException`

nit: remove the new line between `@param` and `@throws` block.